sys/uri_parser/uri_parser.c

```diff
@@ -103,22 +110,46 @@ bool _consume_port(uri_parser_result_t *result, const char *ipv6_end,
 
     /* check if match */
     if (port_begin && (port_begin[0] == ':')) {
-        /* port should be at least one character, => + 1 */
-        if (port_begin + 1 == authority_end) {
+        port_begin++;
+
+        /* port should be at least one character long */
+        if (port_begin == authority_end) {
+            return false;
+        }
+
+        port_str_len = authority_end - port_begin;
+
+        /* Verify that the port number is up to 5 (random) chars in size */
+        if (port_str_len > MAX_PORT_STR_LEN) {
+            return false;
+        }
+
+        /* Verify that all characters of the port are numerical */
+        for (int i = 0; i < port_str_len; ++i) {
+            if (!((port_begin[i] >= '0') && (port_begin[i] <= '9'))) {
+                return false;
+            }
+        }
+
+        /* Verify that the next character, after the port, is an invalid
+         * character for the atol function. Preventing it from reading out-
+         * side of the port section */
+        if ((authority_end[0] >= '0') && (authority_end[0] <= '9')) {
             return false;
         }
-        result->port = port_begin + 1;
-        result->port_len = authority_end - result->port;
 
-        /* ports can't exceed 5 characters */
-        if (result->port_len > 5){
-            result->port = NULL;
-            result->port_len = 0;
+        /* Verify that the port is smaller or equal to UINT16_MAX. */
+        uint32_t port = atol(port_begin);
```

Comment:
there is `strtol` in libc, which has a `endptr`, which is input (until here) and output if `endptr != &authority_end[0]` there was another char in that number, that spare use the for loop checking for in number range and the end check. (l127 to 139)

Reply:
<details><summary>man strtoul</summary>
The strtoul() function converts the initial part of the string in nptr to an unsigned long int value according to the given base, which must be between 2 and 36 inclusive, or be the special value 0.

The string may begin with an arbitrary amount of white space (as determined by [isspace](https://linux.die.net/man/3/isspace)(3)) followed by a single optional '+' or '-' sign. If base is zero or 16, the string may then include a "0x" prefix, and the number will be read in base 16; otherwise, a zero base is taken as 10 (decimal) unless the next character is '0', in which case it is taken as 8 (octal).

The remainder of the string is converted to an unsigned long int value in the obvious manner, stopping at the first character which is not a valid digit in the given base. (In bases above 10, the letter 'A' in either upper or lower case represents 10, 'B' represents 11, and so forth, with 'Z' representing 35.)

If endptr is not NULL, strtoul() stores the address of the first invalid character in *endptr. If there were no digits at all, strtoul() stores the original value of nptr in *endptr (and returns 0). In particular, if *nptr is not '\0' but **endptr is '\0' on return, the entire string is valid.
</details>

Reply:
:+1: